scheduler/reconcile.go

```diff
@@ -1221,9 +1236,9 @@ func (a *allocReconciler) reconcileReconnecting(reconnecting allocSet, all alloc
 					})
 				}
 			} else {
-				// The reconnecting allocation is preferred, so stop this
-				// replacement, but avoid re-stopping stopped allocs
-				if replacementAlloc.ClientStatus != structs.AllocClientStatusFailed {
+				// The reconnecting allocation is preferred, so stop any replacements
+				// that are not in server terminal status or stopped already.
+				if _, ok := stop[replacementAlloc.ID]; !ok {
```

Comment:
I think you're right that we need to check if it's not in `stop` already because otherwise we could append the `allocStopResult` twice. If `replacementAlloc.ClientTerminal()`, will it already be in the `stop` set so the conditional is safe to drop as you've done here?

Reply:
None of the client terminal allocs were in the `stop` set during any of my testing of reconnecting clients.  They all needed to be stopped here.